src/components/Alert.tsx

```diff
@@ -21,6 +28,32 @@ const Alert: FC = () => {
     dispatch(alertActionCreator(null))
   }, [alert, dispatch])
 
+  const undoOrRedo = alert?.alertType === AlertType.Undo || alert?.alertType === AlertType.Redo
+  const buttons = undoOrRedo ? (
+    <div style={{ marginTop: '0.5em' }}>
+      <a
+        className='button button-small'
+        style={{ margin: '0.25em' }}
+        {...fastClick(e => {
+          undoShortcut.exec(dispatch, store.getState, e, { type: 'toolbar' })
```

Comment:
Let's dispatch the action-creator with `dispatch(undo())` instead of executing the shortcut.

You can move the [alert logic](https://github.com/cybersemics/em/blob/9ce879fbff5481c6d30a125774d6c21009b23406/src/shortcuts/undo.ts#L29-L31) into the action-creator itself.

Reply:
Implemented here: dfb05d298b02758840dc6e42b5ba7c6957155c19

Reply:
New commit: df0ab4d73668f56b0cd14b9c7f1ce53b1d27cd98